server/services/team/service.ts

```diff
@@ -730,16 +779,16 @@ export class TeamService extends ServiceClass implements ITeamService {
 		return !!await this.TeamMembersModel.removeRolesByTeamIdAndUserId(teamId, userId, roles);
 	}
 
-	async getInfoByName(teamName: string): Promise<Partial<ITeam> | undefined> {
-		return this.TeamModel.findOne({
+	async getInfoByName(teamName: string): Promise<Omit<ITeam, 'usernames'> | null> {
+		return this.TeamModel.findOne<Omit<ITeam, 'usernames'>>({
 			name: teamName,
-		}, { projection: { usernames: 0 } });
+		}, { projection: { usernames: 0 } }) as unknown as Omit<ITeam, 'usernames'>;
 	}
 
-	async getInfoById(teamId: string): Promise<Partial<ITeam> | undefined> {
-		return this.TeamModel.findOne({
+	async getInfoById(teamId: string): Promise<Omit<ITeam, 'usernames'> | null> {
+		return this.TeamModel.findOne<Omit<ITeam, 'usernames'>>({
 			_id: teamId,
-		}, { projection: { usernames: 0 } });
+		}, { projection: { usernames: 0 } }) as unknown as Omit<ITeam, 'usernames'>;
```

Comment:
is this really needed?

```suggestion
		}, { projection: { usernames: 0 } });
```